ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/Committer.scala

```diff
@@ -117,4 +117,18 @@ object Committer {
           }, conf => Some(Some(entry) -> conf))
       }
       .getOrElse(None -> defaultConfig)
+
+  def buildLogEntryWithOptionalRecordTime(
+      recordTime: Option[Timestamp],
+      addSubmissionSpecificEntry: DamlLogEntry.Builder => DamlLogEntry.Builder): DamlLogEntry = {
+    val logEntryBuilder = DamlLogEntry.newBuilder
+    addSubmissionSpecificEntry(logEntryBuilder)
+    setRecordTimeIfAvailable(recordTime, logEntryBuilder)
+    logEntryBuilder.build
+  }
+
+  private def setRecordTimeIfAvailable(
+      recordTime: Option[Timestamp],
+      logEntryBuilder: DamlLogEntry.Builder): Unit =
```

Comment:
Minor: As this method takes a builder, it should again return the builder to enable the callers to chain method calls on the builder.

Reply:
Good point -- will change it in a follow-up PR.